src/overlays/actors/ovl_En_Fall2/z_en_fall2.c

```diff
@@ -28,26 +35,135 @@ const ActorInit En_Fall2_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fall2/func_80C1B8A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fall2/func_80C1B8B4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fall2/func_80C1B8F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fall2/func_80C1B9D4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fall2/EnFall2_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fall2/EnFall2_Draw.s")
+void EnFall2_Init(Actor* thisx, PlayState* play) {
+    EnFall2* this = THIS;
+
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actionFunc = func_80C1B8A4;
+    func_80183430(&this->skeletonInfo, &D_06008898, &D_06005EF4, &this->unk174, &this->unk228, NULL);
+    func_801835EC(&this->skeletonInfo, &D_06005EF4);
+    this->unk2DC = Lib_SegmentedToVirtual(&D_06008840);
+    Actor_SetScale(&this->actor, 0.02f);
+    this->actionFunc = func_80C1B9D4;
+    this->unk2E6 = 0x231;
+    this->unk2E0 = 0.0f;
+}
+
+void EnFall2_Destroy(Actor* thisx, PlayState* play) {
+    EnFall2* this = THIS;
+
+    func_8018349C(&this->skeletonInfo);
+}
+
+void func_80C1B724(f32 arg0) {
+    static u8 sAlphaTableIndex[] = { 0, 2, 0, 0, 4, 0, 4, 4, 2, 0, 0, 0, 0, 0, 0, 2, 4, 4, 4, 2, 0, 4, 2, 2, 0, 4, 4,
+                                     4, 4, 0, 2, 2, 4, 0, 2, 0, 4, 4, 2, 0, 2, 4, 4, 4, 2, 0, 4, 4, 4, 2, 2, 0, 4, 4,
+                                     2, 4, 0, 2, 0, 0, 2, 4, 4, 2, 0, 2, 4, 4, 4, 2, 0, 4, 4, 4, 4, 4, 2, 2, 0, 2, 4,
+                                     0, 2, 0, 0, 2, 4, 4, 2, 0, 2, 4, 4, 4, 2, 0, 0, 2, 2, 4, 4, 4, 4, 4, 4, 2, 0, 2,
+                                     0, 0, 2, 4, 4, 2, 0, 2, 4, 4, 4, 2, 0, 2, 0, 5, 2, 5, 5, 5, 5, 0, 2, 5, 0, 2, 0,
+                                     2, 5, 5, 2, 0, 2, 5, 5, 5, 2, 0, 3, 3, 3, 2, 2, 0, 0, 3, 3, 0, 2, 2, 3, 0, 2, 3,
+                                     3, 2, 0, 2, 3, 3, 3, 2, 0, 2, 2, 2, 2, 0, 2, 2, 2, 2, 2, 0, 0, 2, 0, 2, 2, 2, 2,
+                                     0, 2, 2, 2, 2, 2, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1,
+                                     1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+    Vtx* vertices = Lib_SegmentedToVirtual(&D_06005F10);
+    u8 perVertexAlphaTable[6];
+    s32 i;
+
+    if (arg0 > 1.0f) {
+        arg0 = 1.0f;
+    }
+    if (arg0 < 0.0f) {
+        arg0 = 0.0f;
+    }
+    perVertexAlphaTable[0] = 0;
+    perVertexAlphaTable[1] = (s8)(54.0f * arg0);
+    perVertexAlphaTable[2] = (s8)(104.0f * arg0);
+    perVertexAlphaTable[3] = (s8)(155.0f * arg0);
+    perVertexAlphaTable[4] = (s8)(255.0f * arg0);
+    perVertexAlphaTable[5] = (s8)(205.0f * arg0);
+
+    for (i = 0; i < ARRAY_COUNT(sAlphaTableIndex); i++, vertices++) {
+        vertices->v.cn[3] = perVertexAlphaTable[sAlphaTableIndex[i]];
+    }
+}
+
+void func_80C1B8A4(EnFall2* this, PlayState* play) {
+}
+
+void func_80C1B8B4(Actor* thisx) {
```

Comment:
Is it *required* for this to take `thisx`? It's called from an actionFunc, so it doesn't make much sense for it to take `thisx` and then upcast it to `this`.